src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1175,6 +1180,14 @@
             (map->InExpression {:column (columns->fields query-rec (utils/vector-maybe column))
                                 :subquery (user-node->plan-node query-rec subquery-expression)})
 
+            [["extract" [["function" (f :guard supported-fns) column]] expr]]
+            (let [column (if (= "value" column)
```

Comment:
Perhaps detecting column type == :multi instead is going to be more future proof here?